docs/src/man/iterative_refinement.md

```diff
@@ -11,61 +12,213 @@ Depth = 3
+plot!(x->1., x->x, -0.8, 1.3, line=1, color="black", linestyle=:dash)
+plot!(x->x+1, x->0., 0.0, 0.4, line=1, color="red", linestyle=:solid, arrow=true)
+plot!(x->0., x->x+1, 0.0, 0.4, line=1, color="red", linestyle=:solid, arrow=true)
+plot!(x->-x, x->x+1, -1.2, .2, line=1., color="black", linestyle=:dashdot)
+plot!(x->x+.6, x->x+.6, -.1, .08, line=1, color="red", linestyle=:solid, arrow=true)
+```
 
-## Stopping criteria
+We can instantiate and append this approximation to a fresh `PolygonalOverapproximation`
+object, which is a type that wraps a set and a list of `LocalApproximation`'s.
+The approximation is refinable, since it can be "split" along `ndir`, where `ndir`
+is the direction normal to the line ``(p_1 p_2)`` (shown dash-dotted in the figure),
+providing two approximations which are closer to the given set in Hausdorff distance.
 
-## Examples
 
-Consider in two dimensions a centered unit ball in the 2-norm. Let's consider a
-polygonal approximation with error bound ``ɛ = .01``.
+```@example example_iterative_refinement
+import LazySets.Approximations:PolygonalOverapproximation, addapproximation!
 
-```@example ir
-using LazySets, LazySets.Approximations, Plots
+Ω = PolygonalOverapproximation{Float64}(b)
+p1, d1, p2, d2 = [1.0, 0.0], [1.0, 0.0], [0.0, 1.0], [0.0, 1.0]
+approx_EAST_NORTH = addapproximation!(Ω, p1, d1, p2, d2)
 
-b = Ball2(zeros(2), 1.)
-ɛ = .5
-p = overapproximate(b, ɛ)
+approx_EAST_NORTH.refinable
+```
+
+The associated error is ``\sqrt{2}-1≈0.414213``, which is the distance between
+the point ``q`` and the intersection between the line ``(0 q)`` and the circle.
+Actually this point corresponds to the support vector of the set `b` along `ndir`.
+
+```@example example_iterative_refinement
+approx_EAST_NORTH.err
+```
+
+The refined approximation is computed next.
+
+## Refinement
 
-plot(b, 1e-3, aspectratio=1)  # use 1e-3 for precise definition of borders
-plot!(p, alpha=0.4)
+Basically, the refinement step consists of splitting the local approximation
+``(p_1, p_2, q)`` into two local approximations ``(p_1, s, q')`` and
+``(s, p_2, q'')``, where `s` is the support vector of ``S`` along `ndir`.
+
+To illustrate this, first let's add the remaining three approximations to `Ω`
+along the canonical directions, to build a box overapproximation of `b`.
+
+```@example example_iterative_refinement
+plot(b, 1e-3, aspectratio=1, alpha=0.3)
+
+# initialize box directions
+DIR_EAST, DIR_NORTH, DIR_WEST, DIR_SOUTH = [1., 0.], [0., 1.], [-1., 0.], [0., -1.]
+pe, pn, pw, ps = σ(DIR_EAST, b), σ(DIR_NORTH, b), σ(DIR_WEST, b), σ(DIR_SOUTH, b)
+
+addapproximation!(Ω, pn, DIR_NORTH, pw, DIR_WEST)
+addapproximation!(Ω, pw, DIR_WEST, ps, DIR_SOUTH)
+addapproximation!(Ω, ps, DIR_SOUTH, pe, DIR_EAST)
+
+plot!(Approximations.tohrep(Ω), alpha=0.2, color="orange")
+```
+
+Next we refine the first approximation of the list.
+
+```@example example_iterative_refinement
+import LazySets.Approximations.refine
```

Comment:
Maybe do `import LazySets.Approximations: refine, tohrep`?

Reply:
done

Reply:
But also remove the two `Approximations.tohrep` 😉